modules/rates-resolver/src/main/scala/fi/spectrumlabs/rates/resolver/services/ResolverService.scala

```diff
@@ -26,23 +26,27 @@ object ResolverService {
     pools: PoolsService[F],
     network: Network[F],
     metadataService: MetadataService[F],
+    tokenFetcher: TokenFetcher[F],
     logs: Logs[I, F]
   ): I[ResolverService[F]] =
     logs.forService[ResolverService[F]].map(implicit __ => new Impl[F](config))
 
   final private class Impl[F[_]: Monad: Logging](config: ResolverConfig)(implicit
     pools: PoolsService[F],
     network: Network[F],
-    metadataService: MetadataService[F]
+    metadataService: MetadataService[F],
+    tokenFetcher: TokenFetcher[F]
   ) extends ResolverService[F] {
 
     def resolve: F[List[ResolvedRate]] =
       network.getAdaPrice
         .flatMap { adaPrice =>
           (for {
-            pools <- pools.getAllLatest(config.minLiquidityValue)
-            info  <- metadataService.getTokensMeta(pools.flatMap(p => p.x.asset :: p.y.asset :: Nil))
-          } yield (pools, info)).map { case (pools, info) =>
+            pools       <- pools.getAllLatest(config.minLiquidityValue)
+            validTokens <- tokenFetcher.fetchTokens
```

Comment:
Looks like it's incorrect to resolve only validated tokens. We should resolve all tokes we can.